File: problem 2.rb
# Problem 2
# By considering the terms in the Fibonacci sequence whose values do not exceed four million, find the sum of the even-valued terms.

first_num = 1
second_num = 2
sum_num = 0
output_arr = []
output_num = 2

while sum_num < 4000000
  sum_num = first_num + second_num
  first_num = second_num
  second_num = sum_num
  if sum_num % 2 == 0
    output_arr.push(sum_num)
  end
end

output_arr.each do |x|
  output_num += x
end

p output_arr
p output_num

File: Problem 2.js
// Problem 2
// By considering the terms in the Fibonacci sequence whose values do not exceed four million, find the sum of the even-valued terms.

valueOne = 1;
valueTwo = 2;
outPutValue = 2;
while(valueTwo < 4000000){
	tempValue = valueOne;
	valueOne = valueTwo;
	valueTwo = tempValue + valueTwo;
	if(valueTwo % 2 === 0 && valueTwo < 4000000){
		outPutValue = outPutValue + valueTwo;
	}
}

console.log(outPutValue);